ruby_on_rails/app_initialisation.md

```diff
@@ -1,61 +1,77 @@
 You may want to choose a different database from Postgres, but most of the time that will be your choice.
 If you do not need a DB you may rethink the fact that you may not need Rails at all :) Take a look at [Sinatra](http://www.sinatrarb.com/) or [Angular](https://angular.io/)
 
-* Specify the ruby version in the project's `Gemfile` with
+* Update the Ruby version in `.ruby-version` and load it in the fresh project's `Gemfile` like this
 
-```ruby File.read(File.join(__dir__, '.ruby-version'))```
+  ```ruby File.read(File.join(__dir__, '.ruby-version'))```
 
-[This is used by Heroku to determine what version to use.](https://devcenter.heroku.com/articles/ruby-versions)
+  [This is used by Heroku to determine what version to use.](https://devcenter.heroku.com/articles/ruby-versions)
 
-## bin scripts
+* Run `bin/setup`
+* Then check your default Rails setup by running `rails s` and visiting `localhost:3000`.
+  You should be on Rails now, yay!
 
-* Add a `bin/run` file which runs `bundle exec puma` and make it executable:
+## Adjustments
 
-```sh
-echo "#\!/bin/sh\n\nbundle exec puma" >> bin/run
-chmod +x bin/run
-```
+### Convenience scripts (`bin/*`)
 
-This will be the file to start our project.
+* Add a `bin/run` file. It will be used to start our project.
 
-* Add an empty `bin/fastcheck` file and make it executable:
+  ```sh
+  echo "#\!/bin/sh\n\nbundle exec puma" >> bin/run
+  ```
 
-```sh
-echo "#\!/bin/sh\n" >> bin/fastcheck
-chmod +x bin/fastcheck
-```
+* Add a `bin/fastcheck` file. It will be used as a hook before pushing to quickly check for linting issues.
 
-This will be used as a hook before pushing to quickly check for linting issues.
+  ```sh
+  echo "#\!/bin/sh\n" >> bin/fastcheck
+  ```
 
-* Add a `bin/check` file which runs `bin/fastcheck` and `bundle exec rake test` and make it executable (`chmod +x bin/check`)
+* Add a `bin/check` file. It will run all the automated tests. It's mainly used in our CI.
 
-```sh
-#!/bin/sh
+  ```sh
+  echo "#\!/bin/sh\nset -e\nbin/fastcheck\n" > bin/check
+  ```
 
-set -e
+* Make the new scripts executable
 
-bin/fastcheck
+  ```sh
+  chmod +x bin/run bin/fastcheck bin/check
+  ```
 
-bundle exec rake test
-```
+* `bin/setup` was generated by Rails. We'll add a `pre-push` hook to it:
 
-This will be the command that runs the tests for our applications.
-We use a script for that so that we can change the commands for tests in a PR, without breaking the CI and without the need of changing its configuration.
+  ```ruby
+  unless File.exist?('.git/hooks/pre-push')
+    system 'ln -s ../../bin/fastcheck .git/hooks/pre-push'
+  end
+  ```
+
+  Also change `system!('bundle install')` to `system!('bundle install --jobs=3 --retry=3')`
+  and uncomment the installation of yarn dependencies with `system('bin/yarn')`
+
+* Run `bundle exec rails db:migrate` to generate an empty `schema.rb` file
+* Run `bin/setup`.
```

Comment:
do I need to check something afterwards? Otherwise I would ask myself, why running it again